checker/src/org/checkerframework/checker/upperbound/UpperBoundStore.java

```diff
@@ -0,0 +1,157 @@
+package org.checkerframework.checker.upperbound;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.checker.upperbound.qual.LTEqLengthOf;
+import org.checkerframework.checker.upperbound.qual.LTLengthOf;
+import org.checkerframework.checker.upperbound.qual.LTOMLengthOf;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.FieldAccess;
+import org.checkerframework.dataflow.analysis.FlowExpressions.LocalVariable;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.framework.flow.CFAbstractAnalysis;
+import org.checkerframework.framework.flow.CFAbstractStore;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+public class UpperBoundStore extends CFAbstractStore<CFValue, UpperBoundStore> {
+
+    protected UpperBoundStore(UpperBoundStore other) {
+        super(other);
+    }
+
+    public UpperBoundStore(
+            CFAbstractAnalysis<CFValue, UpperBoundStore, ?> analysis, boolean sequentialSemantics) {
+        super(analysis, sequentialSemantics);
+    }
+
+    // If something is removed from a list it reduces the minlen of anything that could be an alias of the list by 1.
+    // If a list is cleared, anything that could be an alias of this list goes to UpperBoundUnknown.
```

Comment:
These comments are out-of-date.  Instead of a comment here.  I would add a comment for each if block in the method itself.